aws-predictions/src/test/java/com/amplifyframework/predictions/aws/http/LivenessWebSocketTest.kt

```diff
@@ -317,6 +325,57 @@ internal class LivenessWebSocketTest {
         assertEquals(livenessWebSocket.getUserAgent(), "$baseline $additional")
     }
 
+    @Test
+    fun `web socket detects clock skew from server response`() {
+        val livenessWebSocket = createLivenessWebSocket()
+        mockkConstructor(WebSocket::class)
+        val socket: WebSocket = mockk()
+        livenessWebSocket.webSocket = socket
+        val sdf = SimpleDateFormat(livenessWebSocket.datePattern, Locale.getDefault())
+
+        // server responds saying time is actually 1 hour in the future
+        val oneHour = 1000 * 3600
+        val futureDate = sdf.format(Date(Date().time + oneHour))
+        val response: Response = mockk()
+        every { response.headers }.returns(Headers.headersOf("Date", futureDate))
+        every { response.header("Date") }.returns(futureDate)
+        livenessWebSocket.webSocketListener.onOpen(socket, response)
+
+        // now we should restart the websocket with an adjusted time
+        val openLatch = CountDownLatch(1)
+        val latchingListener = LatchingWebSocketResponseListener(
+            livenessWebSocket.webSocketListener,
+            openLatch = openLatch
+        )
+        livenessWebSocket.webSocketListener = latchingListener
+
+        server.enqueue(MockResponse().withWebSocketUpgrade(ServerWebSocketListener()))
+        server.start()
+        livenessWebSocket.start()
+
+        openLatch.await(3, TimeUnit.SECONDS)
+
+        assertTrue(livenessWebSocket.webSocket != null)
+        val originalRequest = livenessWebSocket.webSocket!!.request()
+
+        // make sure that followup request sends offset date
+        val sdfGMT = SimpleDateFormat("yyyyMMdd'T'HHmmss'Z'", Locale.getDefault())
+        sdfGMT.timeZone = TimeZone.getTimeZone("GMT")
+        val sentDate = originalRequest.url.queryParameter("X-Amz-Date") ?.let { sdfGMT.parse(it) }
+        val diff = abs(Date().time - sentDate?.time!!)
+        assert(oneHour - 10000 < diff && diff < oneHour + 10000)
+
+        // also make sure that followup request is valid
+        assertEquals("AWS4-HMAC-SHA256", originalRequest.url.queryParameter("X-Amz-Algorithm"))
+        assertTrue(
+            originalRequest.url.queryParameter("X-Amz-Credential")!!.endsWith("//rekognition/aws4_request")
+        )
+        assertEquals("299", originalRequest.url.queryParameter("X-Amz-Expires"))
+        assertEquals("host", originalRequest.url.queryParameter("X-Amz-SignedHeaders"))
+        assertEquals("AWS4-HMAC-SHA256", originalRequest.url.queryParameter("X-Amz-Algorithm"))
+        assertNotNull("x-amz-user-agent")
```

Comment:
The assertion does not appear to test anything as it is passing a string literal instead of a variable or a function call result. Maybe you intended to test the query parameter in orignalRequest Object here.